extensions/amp-date-display/0.2/render-dom-tree.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpEvents} from '../../../src/amp-events';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dev, devAssert} from '../../../src/log';
+import {removeChildren} from '../../../src/dom';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * Clears the host element and appends the DOM tree into it.
+ *
+ * @param {!JsonObject} props
+ * @return {null}
+ */
+export function RenderDomTree(props) {
```

Comment:
Should we move this to `src/preact/` somewhere along with `AsyncRender`? These seem to be important classes to support a particular type of our components.